### DISCUSSION TITLE
converting an RGB image to rec709YCbCr and moving Cb and Cr by half a subpixel to the left and back to RGB

### DISCUSSION THREAD

**Discussion by mightyhuhn** [ImageMagick/ImageMagick#7744] [Help] [answered]
### ImageMagick version

7.1.1-q16-HDRI

### Operating system, version and so on

win 10 2022h2

### Description

just to be absolutely clear google has the answer nearly for sure to this question the page is sadly http error 500.

what i already got is this: 
magick.exe "testimage.png" -colorspace Rec709YCbCr -distort srt "0,0 1 0 0.0,-0.5" -colorspace RGB "newtestimage.png"

the problem is i don't know how to limited imagemagick to Cb and Cr.
the distort numbers are nearly for sure wrong too. try and error should get me there.
i most likely need to define that the image is just gamma too.
moving it by half a subpixel and also upscaling to for example 1080p to UHD would also be interesting to lower the scaling error.

**Comment by snibgo** [comment, marked as answer]
Those SRT numbers will shift by half a pixel *down*. You want to shift to the *left*, so invert the last two numbers. See https://imagemagick.org/script/command-line-options.php#distort
```
magick in.png -colorspace Rec709YCbCr ( +clone -distort srt "0,0 1 0 -0.5,0" ) -separate -delete 1-3 +combine Rec709YCbCr -colorspace sRGB out.png
```
This makes a clone of the image, and shifts that by 0.5 pixels. Now we have two channels. `-separate` converts each image to three grayscale images, one per channel per image. So now we have 5 channels, representing Y,Cb,Cr,Y,Cb,Cr. Delete the image numbers 1, 2 and 3 to remove the Cb and Cr from the first image and Y from the second image. So now we have Y,Cb and Cr. Combine them. Convert them to sRGB colorspace. (Or convert to linear RGB if you prefer.)

> i most likely need to define that the image is just gamma too.
moving it by half a subpixel and also upscaling to for example 1080p to UHD would also be interesting to lower the scaling error.

I don't understand that.
